src/Chirp.Razor/Pages/Public.cshtml.cs

```diff
@@ -8,16 +8,27 @@ namespace Chirp.Razor.Pages;
 public class PublicModel : PageModel
 {
     private readonly ICheepService _service;
+    [FromQuery(Name = "page")]
+    public string? QueryPage { get; set; }
+    public int PageNumber { get; set; }
+    public Range CheepRange {get;set;}
     public List<CheepViewModel> Cheeps { get; set; }
 
     public PublicModel(ICheepService service)
     {
         _service = service;
+        Cheeps = _service.GetCheeps();
+
     }
 
     public ActionResult OnGet()
     {
-        Cheeps = _service.GetCheeps();
+        PageNumber = int.TryParse(QueryPage, out var page) && 0<=page && page<=MathUtil.PageAmount(Cheeps) ? page : 1;
```

Comment:
Could use some formatting in this section, probably other places too

Reply:
> Could use some formatting in this section, probably other places too

I will look into that 👍

Reply:
Maybe #68f8c58f29f9a3dcae5ca433ecbc23c2661fd4e9 resolves this?

Reply:
Maybe 68f8c58f29f9a3dcae5ca433ecbc23c2661fd4e9 solves it?